src/artofillusion/polymesh/QuadMesh.java

```diff
@@ -1278,6 +1285,11 @@ private void smoothMesh(double tol, boolean calcProjectedEdges, int ns, int maxN
             moveVerts[vertices.length + index] = false;
             v1 = edges[i].v1;
             v2 = edges[i].v2;
+            if (v1 == -1 || v2 == -1)
+                // I don't know if this is even theoretically possible, but 
+                // strange things have happened with old PolyMesh models.
+                continue;
```

Comment:
So, if I understand correctly, this is a guard for an error condition that we don't have any live examples for?

Reply:
A similar structure has caused problem elsewhere. 

It is pretty typical Java that "-1" stands for "no value" in index variables. When that is used, the -1-case should be handled somewhere in the chain. Somehow the `if (x == -1)` seem lighter to me than `try-catch-throw...`